pkg/apply/prune/prune_test.go

```diff
@@ -259,35 +265,57 @@ func TestPrune(t *testing.T) {
 						DryRunStrategy: drs,
 					})
 				}()
-				if !tc.isError {
-					if err != nil {
-						t.Fatalf("Unexpected error during Prune(): %#v", err)
-					}
 
-					var actualPruneEvents []event.Event
-					for e := range eventChannel {
-						actualPruneEvents = append(actualPruneEvents, e)
-					}
-					if want, got := len(tc.pruneEventObjs), len(actualPruneEvents); want != got {
-						t.Errorf("Expected (%d) prune events, got (%d)", want, got)
-					}
+				if err != nil {
+					t.Fatalf("Unexpected error during Prune(): %#v", err)
+				}
+
+				// Test that the correct inventory objects are stored at the end of the prune.
+				actualObjs := fakeInvClient.Objs
+				expectedObjs := object.UnstructuredsToObjMetas(tc.finalClusterObjs)
+				if !object.SetEquals(expectedObjs, actualObjs) {
+					t.Errorf("expected inventory objs (%s), got (%s)", expectedObjs, actualObjs)
+				}
+
+				var actualPruneEvents []event.Event
+				for e := range eventChannel {
+					actualPruneEvents = append(actualPruneEvents, e)
+				}
+				if want, got := len(tc.pruneEventObjs), len(actualPruneEvents); want != got {
+					t.Errorf("Expected (%d) prune events, got (%d)", want, got)
+				}
 
-					for i, obj := range tc.pruneEventObjs {
-						e := actualPruneEvents[i]
-						expKind := obj.GetObjectKind().GroupVersionKind().Kind
-						actKind := e.PruneEvent.Identifier.GroupKind.Kind
-						if expKind != actKind {
-							t.Errorf("Expected kind %s, got %s", expKind, actKind)
-						}
+				for i, obj := range tc.pruneEventObjs {
+					e := actualPruneEvents[i]
+					expKind := obj.GetObjectKind().GroupVersionKind().Kind
+					actKind := e.PruneEvent.Identifier.GroupKind.Kind
+					if expKind != actKind {
+						t.Errorf("Expected kind %s, got %s", expKind, actKind)
 					}
-				} else if err == nil {
-					t.Fatalf("Expected error during Prune() but received none")
 				}
 			})
 		}
 	}
 }
 
+// unionObjects returns the union of sliceA and sliceB as a slice of unstructured objects.
```

Comment:
If both sliceA and sliceB contains the unstructured object with the same ObjMeta, the one from sliceB is taken.